ir/src/constraint_builder/mod.rs

```diff
@@ -1,3 +1,5 @@
+use std::fmt;
+
```

Comment:
Let's remove the extra line here and also import the items below from super instead of from `constraints`